### PR TITLE
Add an example of secret injection in legacy config file

### PR DESCRIPTION
**Pull request by mkmik**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -466,10 +466,19 @@ $ kubectl replace -f master.key
 $ kubectl delete pod -n kube-system -l name=sealed-secrets-controller
 ```
 
-- What flags are available for kubeseal?
+### What flags are available for kubeseal?
 
 You can check the flags available using `kubeseal --help`.
 
+### How do update parts of JSON/YAML/TOML.. file encrypted with sealed secrets?
```
Comment (atomatt) [inline README.md]: s/do/do I/s ?

File: docs/examples/config-template/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Injecting secrets in config file templates
+
+Kubernetes Secrets are very flexible and can be consumed in many ways.
+Secret values can be passed to containers as environment variables or appear as regular files when mouting secret volumes.
```
Comment (atomatt) [inline docs/examples/config-template/README.md]: s/mouting/mounting/

File: docs/examples/config-template/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+Often users end up using the latter method just to wrap full configuration files into k8s secrets, just
+because one or more fields in the config file happen to be secrets (e.g. a database password, or a session cookie encryption key).
+
+Ideally you should avoid configuring your software that way and instead splitting your configuration from your secrets somehow. Also make sure you know about [12 Factor](https://www.12factor.net/).
```
Comment (atomatt) [inline docs/examples/config-template/README.md]: s/ and instead/, instead/ or s/splitting/split/

**Comment by mkmik** [comment]
thanks (I tested the k8s files); yeah it's technically outside of the scope but this is literally an FAQ so from the POV of users it must be in scope :-)

**Comment by atomatt** [comment]
Yeah, I assumed it was a common enough question for you to bother with it :).

File: docs/examples/config-template/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  template:
+    metadata:
+      labels:
+        app: example
+    spec:
+      initContainers:
+      - name: inject-secrets
+        image: hairyhenderson/gomplate:alpine
+        volumeMounts:
+        # The /secrets directory will contain one file per sercret item.
```
Comment (dbyron0) [inline docs/examples/config-template/deployment.yaml]: s/sercret/secret

**Pull request merged**